mllib/src/main/scala/org/apache/spark/mllib/feature/StandardScaler.scala

```diff
@@ -61,20 +61,34 @@ class StandardScaler(withMean: Boolean, withStd: Boolean) extends Logging {
  * :: Experimental ::
  * Represents a StandardScaler model that can transform vectors.
  *
- * @param withMean whether to center the data before scaling
- * @param withStd whether to scale the data to have unit standard deviation
  * @param mean column mean values
  * @param variance column variance values
+ * @param withMean whether to center the data before scaling
+ * @param withStd whether to scale the data to have unit standard deviation
  */
 @Experimental
-class StandardScalerModel private[mllib] (
-    val withMean: Boolean,
-    val withStd: Boolean,
+class StandardScalerModel (
     val mean: Vector,
-    val variance: Vector) extends VectorTransformer {
+    val variance: Vector,
+    var withMean: Boolean,
+    var withStd: Boolean) extends VectorTransformer {
 
   require(mean.size == variance.size)
 
+  def this(mean: Vector, variance: Vector) {
+    this(mean, variance, false, true)
+  }
+
```

Comment:
I have a question about this API. If the default `withMean` is `false`, why do we require `mean` in the constructor? If the feature dimension is really large, this puts some extra cost that cannot be ignored. Similarly, should we take std directly instead of variance in the constructor? My proposal is the following:

* `StandoardScalerModel(std: Vector, mean: Vector, withStd: Boolean, withMean: Boolean)`.  I put `variance` in front of `mean` because scaling is used more frequently than shifting.
* `this(std: Vector, mean: Vector)`: enable `withMean` and `withStd` based on whether the input arguments are null or not. Throw exception is both are null.
* `this(std: Vector) = this(std, null)`.
* `setWithMean` and `setWithStd` check whether the corresponding `mean`/`variance` is null or not and throw exceptions if a user want to set it to true while the value is null.